### ISSUE TITLE
After Upgrading To 1.7.12 i have gotten Error 520

### ISSUE THREAD

**Issue by MaximKing1** [litespeedtech/openlitespeed#256]
After upgrading to the 1.7.12 and on some apps i get a CloudFlare Error 520?? Any ideas cant even access the web dash as i get the Error 520. I downgraded and works fine

**Comment by litespeedtech** [comment]
You need to add CloudFlare IP as trusted IP, as 1.7.12 fixed the broken connection throttling limit, if CF server make too many connections, it will be blocked. 
https://openlitespeed.org/kb/show-real-visitor-ip-instead-of-cloudflare-ips/

**Comment by MaximKing1** [comment]
Thanks, i had that set to ALL before but in case anyone else needs it
```
173.245.48.0/20T, 103.21.244.0/22T, 103.22.200.0/22T, 103.31.4.0/22T, 141.101.64.0/18T, 108.162.192.0/18T, 190.93.240.0/20T, 188.114.96.0/20T, 197.234.240.0/22T,198.41.128.0/17T, 162.158.0.0/15T, 172.64.0.0/13T, 131.0.72.0/22T, 104.16.0.0/13T, 104.24.0.0/14T
```

**Comment by MaximKing1** [comment]
> You need to add CloudFlare IP as trusted IP, as 1.7.12 fixed the broken connection throttling limit, if CF server make too many connections, it will be blocked.
> https://openlitespeed.org/kb/show-real-visitor-ip-instead-of-cloudflare-ips/

Im still having this issue and added the stuff i added above..

**Comment by litespeedtech** [comment]
Anything in the error log?

**Comment by MaximKing1** [comment]
> Anything in the error log?

No, I can’t find anything in there but just shows an Error 520 when even when I add the CloudFlare IP’s as trusted… what’s weird it works for a bit then stops after 10 requests.

**Comment by litespeedtech** [comment]
It should log the reason why an IP was blocked. 
Maybe you need to enable debug logging.

**Comment by MaximKing1** [comment]
> It should log the reason why an IP was blocked.
> Maybe you need to enable debug logging.

I have enabled debug logging, it works for a little bit then it just shows that Error 520? Any ideas

**Comment by litespeedtech** [comment]
I need to check the error log. you can send email to bug _at_ litespeedtech.com

**Comment by MaximKing1** [comment]
> I need to check the error log. you can send email to bug {at} litespeedtech.com

Let me do some more testing and see if I can get it, do you know where the file is? When I change the version back the error log looks likes it resets or ever has so many it can’t show them all.

**Comment by Code-Egg** [comment]
If it's blocked by the throttling, should see something like this from logs/error.log
```
2021-07-09 07:21:27.480626 [NOTICE] [6338] [x.x.x.x] Reached per client hard connection limit: 1, current: 1, close connection!
2021-07-09 07:21:27.480799 [WARN] [6338] [x.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
```

**Comment by MaximKing1** [comment]
I’ve found this
```


[253871] [HIDDEN] bot detected for vhost [N/A], reason: OverConnSoftLimit, close connection!
--
2021-07-07 16:58:50.194356 | WARN | [253869] [HIDDEN] bot detected for vhost [N/A], reason: OverConnSoftLimit, close connection!
2021-07-07 16:59:11.517373 | WARN | [253871] [HIDDEN:31368#OpenLiteSpeed] Invalid Http response header, retry!
2021-07-07 16:59:11.518452 | WARN | [253871] [NOT-STEALING-MY-IP-TODAY:31368#OpenLiteSpeed] Invalid Http response header, retry!
2021-07-07 16:59:11.519525 | WARN | [253871] [NOPE:31368#OpenLiteSpeed] Invalid Http response header, retry!

```

**Comment by litespeedtech** [comment]
Your connection soft limit may set too low. You should increase it. 
Is it a CloudFlare IP or your IP?

**Comment by MaximKing1** [comment]
> Your connection soft limit may set too low. You should increase it.
> Is it a CloudFlare IP or your IP?

I’ve added every CloudFlare IP as trusted, and my soft limit is at 15 as it suggests in the DDOS limit settings.. What should I increase to?

**Comment by litespeedtech** [comment]
try 50, then gradually lower it if need.

**Comment by MaximKing1** [comment]
> try 50, then gradually lower it if need.

But shouldn’t it be fine anyway? It should ignore trusted IP’s

**Comment by litespeedtech** [comment]
We need the notice level logging to tell if a trusted IP is ignored or not. 
Please lower the log level, to info, or debug.
To really tell what is going on, need debug level logging.

**Comment by MaximKing1** [comment]
> We need the notice level logging to tell if a trusted IP is ignored or not.
> Please lower the log level, to info, or debug.
> To really tell what is going on, need debug level logging.

Okay, I fixed that issue seems like I had a massive ish DDoS attack and now when I turn on OpenLiteSpeed on one of my servers it shows it can’t use Port 80 or 443 as a listener? Any ideas

**Comment by litespeedtech** [comment]
So, it is a configuration issue? 
I am not sure what is the error, but, you need to stop all other server using port 80 and 443.
you can check with 
`netstat -anp | grep LISTEN` 
see which process takes the port 80 and 443.

**Comment by MaximKing1** [comment]
> So, it is a configuration issue?
> I am not sure what is the error, but, you need to stop all other server using port 80 and 443.
> you can check with
> `netstat -anp | grep LISTEN`
> see which process takes the port 80 and 443.

None show Port 80 or 443?

**Comment by litespeedtech** [comment]
Maybe it is a configuration mistake, tries to use port 80 and 443 for webadmin vhost?
What is the exact error message?

**Comment by MaximKing1** [comment]
[12625] No listener is available, stop server!

**Comment by litespeedtech** [comment]
If you configured listener for port 80 and 443, there must be problem with that configuration, please check error messages before this log.

**Comment by litespeedtech** [comment]
Looks those are configuration issues, not server bug. case closed.

**Comment by MaximKing1** [comment]
This is a server Bug!

**Comment by MaximKing1** [comment]
@litespeedtech Please Reopen This, 3 People Using OLS Have Gotten Same Errors As Me!

**Comment by elitetheespeon** [comment]
I have this same issue after updating and it seems to stem from the "bot detection"

```
[root@web ~]# grep "bot detected for vhost" /usr/local/lsws/logs/error.log
2021-07-14 14:24:12.391111 [WARN] [1496630] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-14 14:25:50.973919 [WARN] [1496631] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-15 01:28:16.203519 [WARN] [1542909] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-15 01:28:58.269004 [WARN] [1542908] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-15 16:43:05.884744 [WARN] [1579410] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-15 16:46:22.711099 [WARN] [1579409] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-16 12:52:39.615067 [WARN] [1655988] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-16 12:53:50.949418 [WARN] [1655989] [128.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-19 09:34:19.766764 [WARN] [1689692] [73.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
2021-07-19 10:11:48.796451 [WARN] [1689691] [73.x.x.x] bot detected for vhost [N/A], reason: OverConnHardLimit, close connection!
```

I have not changed any of my config, just updated OpenLiteSpeed. Was there any change made to bot detection and/or limits?

**Comment by MaximKing1** [comment]
> o

Yes, im getting this aswell. @litespeedtech Please reopen this...

**Comment by litespeedtech** [comment]
It has been fixed in the latest update of 1.7.12. Please do 
/usr/local/lsws/admin/misc/lsup.sh -e 1.7.12
If installed through repo, please do package update.
